**1. Import Data**

Data diakses dari github, lalu di sambungkan ke google colabs menggunakan fungsi pd.read_csv, data yang diakses ada beberapa kota yaitu kota Jakarta, Bogor, dan Bekasi.

In [ ]:
# Loading Data:
import pandas as pd
import matplotlib
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
import gc
import numpy as np
from collections import Counter
matplotlib.rcParams['figure.dpi'] = 100
sns.set()
%matplotlib inline
try: # Running Locally, yakinkan "file_" berada di folder "data"
    jakarta = pd.read_csv("data/jakarta.csv", low_memory = False, encoding='utf8')
    bogor = pd.read_csv("data/bogor.csv", low_memory = False, encoding='utf8')
    bekasi = pd.read_csv("data/bekasi.csv", low_memory = False, encoding='utf8')
except: # Running in Google Colab
    !mkdir data
    !wget -P data/ https://raw.githubusercontent.com/taufiksutanto/EDA-Case-01-PreProcessing/master/data/jakarta.csv
    !wget -P data/ https://raw.githubusercontent.com/taufiksutanto/EDA-Case-01-PreProcessing/master/data/bogor.csv
    !wget -P data/ https://raw.githubusercontent.com/taufiksutanto/EDA-Case-01-PreProcessing/master/data/bekasi.csv
    jakarta = pd.read_csv("data/jakarta.csv", low_memory = False, encoding='utf8')
    bogor = pd.read_csv("data/bogor.csv", low_memory = False, encoding='utf8')
    bekasi = pd.read_csv("data/bekasi.csv", low_memory = False, encoding='utf8')
    
jakarta.shape, bogor.shape, bekasi.shape

((1024, 26), (999, 26), (992, 26))

Terdapat hasil jumlah data baris dan kolom data yang belum dibersihkan dan dirapikan. Yaitu 1024 baris dan 26 kolom untuk kota Jakarta, 999 baris dan 26 kolom untuk kota Bogor, dan 992 baris dan 26 kolom untuk kota Bekasi.

In [ ]:
jakarta = pd.DataFrame(jakarta)
bogor = pd.DataFrame(bogor)
bekasi = pd.DataFrame(bekasi)

In [ ]:
jakarta.info(), bogor.info(), bekasi.info()

In [ ]:
jakarta.describe().transpose(), 

In [ ]:
bogor.describe().transpose()

In [ ]:
bekasi.describe().transpose()

Melakukan statistika deskriptif masing masing kota untuk menggambarkan jumlah, rata rata, standar deviasi, kuartil 1, 2, dan 3, minimal serta maksimal. Masing masing komponen mmemberikan gambaran mengenai variabel atau data yang digunakan dan diproses.

**2. Data Prepocessing**

In [ ]:
#menghapus kolom yang berisi data NaN
jakarta.drop(["created_at", "hadap", "Unnamed: 14",'Unnamed: 14', 'Unnamed: 15', 'Unnamed: 16', 'Unnamed: 17','Unnamed: 18', 'Unnamed: 19', 'Unnamed: 20', 'Unnamed: 21', 'Unnamed: 22', 'Unnamed: 23', 'Unnamed: 24', 'Unnamed: 25'], axis=1, inplace=True)


In [ ]:
bogor.drop(["created_at", "hadap","Unnamed: 14",'Unnamed: 14', 'Unnamed: 15', 'Unnamed: 16', 'Unnamed: 17','Unnamed: 18', 'Unnamed: 19', 'Unnamed: 20', 'Unnamed: 21', 'Unnamed: 22', 'Unnamed: 23', 'Unnamed: 24', 'Unnamed: 25'], axis=1, inplace=True)


In [ ]:
bekasi.drop(["created_at", "hadap","Unnamed: 14",'Unnamed: 14', 'Unnamed: 15', 'Unnamed: 16', 'Unnamed: 17','Unnamed: 18', 'Unnamed: 19', 'Unnamed: 20', 'Unnamed: 21', 'Unnamed: 22', 'Unnamed: 23', 'Unnamed: 24', 'Unnamed: 25'], axis=1, inplace=True)


In [ ]:
jakarta.drop_duplicates(inplace = True)
print(jakarta.shape)
jakarta.duplicated().sum()

In [ ]:
jakarta.isnull().sum()

In [ ]:
bogor.drop_duplicates(inplace = True)
print(bogor.shape)
bogor.duplicated().sum()

In [ ]:
bogor.isnull().sum()

In [ ]:
bekasi.drop_duplicates(inplace = True)
print(bekasi.shape)
bekasi.duplicated().sum()

In [ ]:
bekasi.isnull().sum()

Untuk merapikan data yang berisikan NaN kita menghapus kolom yang berisikan data NaN dengan menggunakan fungsi drop_duplicates. Selanjutnya kita menggunakan fungsi isnull().sum() untuk merubah data Nan menjadi 0

In [ ]:
null = jakarta["KT"].isnull()
jakarta["KT"] = jakarta["KT"].replace(np.nan, 0)
null = jakarta["KM"].isnull()
jakarta["KM"] = jakarta["KM"].replace(np.nan, 0)
null = jakarta["garasi"].isnull()
jakarta["garasi"] = jakarta["garasi"].replace(np.nan, 0)
null = jakarta["carport"].isnull()
jakarta["carport"] = jakarta["carport"].replace(np.nan, 0)
null = jakarta["listrik"].isnull()
jakarta["listrik"] = jakarta["listrik"].replace(np.nan, 0)


mean = jakarta["LB"].mean()
jakarta["LB"] = jakarta["LB"].replace(np.nan, mean)
mean = jakarta["LT"].mean()
jakarta["LT"] = jakarta["LT"].replace(np.nan, mean)
mean = jakarta["harga"].mean()
jakarta["harga"] = jakarta["harga"].replace(np.nan, mean)


jakarta

In [ ]:
null = bogor["KT"].isnull()
bogor["KT"] = bogor["KT"].replace(np.nan, 0)
null = bogor["KM"].isnull()
bogor["KM"] = bogor["KM"].replace(np.nan, 0)
null = bogor["garasi"].isnull()
bogor["garasi"] = bogor["garasi"].replace(np.nan, 0)
null = bogor["carport"].isnull()
bogor["carport"] = bogor["carport"].replace(np.nan, 0)
null = bogor["listrik"].isnull()
bogor["listrik"] = bogor["listrik"].replace(np.nan, 0)


mean = bogor["LB"].mean()
bogor["LB"] = bogor["LB"].replace(np.nan, mean)
mean = bogor["LT"].mean()
bogor["LT"] = bogor["LT"].replace(np.nan, mean)
mean = bogor["harga"].mean()
bogor["harga"] = bogor["harga"].replace(np.nan, mean)


bogor

In [ ]:
null = bekasi["KT"].isnull()
bekasi["KT"] = bekasi["KT"].replace(np.nan, 0)
null = bekasi["KM"].isnull()
bekasi["KM"] = bekasi["KM"].replace(np.nan, 0)
null = bekasi["garasi"].isnull()
bekasi["garasi"] = bekasi["garasi"].replace(np.nan, 0)
null = bekasi["carport"].isnull()
bekasi["carport"] = bekasi["carport"].replace(np.nan, 0)
null = bekasi["listrik"].isnull()
bekasi["listrik"] = bekasi["listrik"].replace(np.nan, 0)


mean = bekasi["LB"].mean()
bekasi["LB"] = bekasi["LB"].replace(np.nan, mean)
mean = bekasi["LT"].mean()
bekasi["LT"] = bekasi["LT"].replace(np.nan, mean)
mean = bekasi["harga"].mean()
bekasi["harga"] = bekasi["harga"].replace(np.nan, mean)


bekasi

***Interpretasi dan Jawaban***

1. Kualitas data yang dimiliki perusahaan.

>> Setelah memproses data yang telah diberikan, kualitas data yang dimiliki perusahaan cenderung kotor dan banyak data yang kosong. Ada beberapa data yang kurang membantu, sehingga data yang diberikan tersebut memiliki kualitas yang buruk.

2. Apakah ada saran perbaikan data ke DE/DBA perusahaan agar RPPI menjadi data driven company yang unggul.

>> Saran yang bisa tim kami sampaikan. DE/DBA bisa mengumpulkan data seakurat dan sedetail mungkin, agar tim kami bisa lebih cepat dan lebih rapih dalam menganalisis data yang diberikan, karena terdapat beberapa data yang kosong, sehingga menyulitkan tim kami dalam melakukan proses analisis data.

3. Dari data yang ada, apakah ada kecenderungan rumah disuatu kota lebih mahal/murah dibandingkan kota lain?

>> Ada, setelah membandingkan dari ketiga kota yang telah kami proses. Kami bisa menyimpulkan bahwa kota jakarta cenderung mahal dari pada 2 kota lainnya dengan rata-rata harga rumah di jakarta berkisar 8 miliar, sedangkan 2 kota lainnya berkisar sekitar 1 miliar, ini bisa dikarenakan luas bangunan atau letak bangunan di jakarta yang strategis.

4. Rumah dengan karakteristik seperti apa yang paling banyak dijual?

>> Rumah dengan fasilitas carport lebih banyak dijual dibandingkan dengan rumah yang memiliki fasilitas garasi. Rumah yang memiliki sertifikat tentu saja lebih menjual daripada rumah yang tidak memiliki sertifikat, namun dari data yang diberikan perusahaan masih ada rumah yang tidak memiliki sertifikat tetapi tetap diperjualbelikan. 

5. Jika RPPI ingin melakukan investasi (max) Rp. 25 Milyar minggu besok, apakah dari data (statistika deskriptif sederhana) anda dapat memberikan rekomendasi rumah mana saja yang akan berpotensi menghasilkan keuntungan bagi perusahaan?

>> Setelah melihat hasilnya, kami bisa memberikan perusahaan beberapa rekomendasi rumah yang sangat berpotensi menghasilkan keuntungan besar bagi perusahaan, berikut datanya

6. Dan yang terpenting, informasi berharga apa lagi yang bisa anda dapatkan dari data melalui hanya preprocessing dan statistika deskriptif?

>> 

**Kesimpulan**

Data yang diberikan oleh perusahaan tidak lengkap atau masih banyak yang kosong, sehingga diperlukan data yang lengkap untuk proses penganalisaan yang baik. Kota Jakarta memiliki harga jual yang paling tinggi diantara harga jual ke 2 kota lainnya. Hal tersebut dikarenakan Jakarta memiliki luas bangunan dan letak yang strategis. Karakteristik rumah yang banyak diminati oleh pembeli adalah yang memiliki carport dan bersertifikat. Sehingga dari data yang kami olah, perusahaan dapat menentukan jenis rumah yang memiliki keuntungan yang lebih besar untuk perusahaan.